coral/src/domain/requests/requests-api.ts

```diff
@@ -1,20 +1,43 @@
-import { getRequestsWaitingForApprovalTransformer } from "src/domain/requests/requests-transformers";
+import {
+  activityLogTransformer,
+  getRequestsWaitingForApprovalTransformer,
+} from "src/domain/requests/requests-transformers";
 import api, { API_PATHS } from "src/services/api";
 import { KlawApiRequestQueryParameters, KlawApiResponse } from "types/utils";
 
-const getRequestsStatistics = (
+const getRequestsStatistics = async (
   params: KlawApiRequestQueryParameters<"getRequestStatistics">
 ): Promise<KlawApiResponse<"getRequestStatistics">> => {
-  return api.get<KlawApiResponse<"getRequestStatistics">>(
+  const response = await api.get<KlawApiResponse<"getRequestStatistics">>(
     API_PATHS.getRequestStatistics,
     new URLSearchParams(params)
   );
+
+  return response;
 };
 
-const getRequestsWaitingForApproval = () => {
-  return getRequestsStatistics({ requestMode: "MY_APPROVALS" }).then(
-    getRequestsWaitingForApprovalTransformer
-  );
+const getRequestsWaitingForApproval = async () => {
+  const response = await getRequestsStatistics({
+    requestMode: "MY_APPROVALS",
+  }).then(getRequestsWaitingForApprovalTransformer);
+
+  return response;
+};
+
+const getActivityLog = async (
+  params: KlawApiRequestQueryParameters<"showActivityLog">
+) => {
+  const response = await api
+    .get<KlawApiResponse<"showActivityLog">>(
+      API_PATHS.showActivityLog,
+      new URLSearchParams({
+        pageNo: params.pageNo,
+        ...(params.env && { env: params.env }),
+      })
+    )
+    .then(activityLogTransformer);
+
+  return response;
 };
 
-export { getRequestsStatistics, getRequestsWaitingForApproval };
+export { getActivityLog, getRequestsStatistics, getRequestsWaitingForApproval };
```

Comment:
smol note: `getActivityLog` needs to be exported (and also imported) from the barrel file in `requests/index.ts`, same with types etc that are used outside of `/domain`